olp-cpp-sdk-dataservice-write/src/VersionedLayerClientImpl.cpp

```diff
@@ -465,195 +465,110 @@ olp::client::CancellationToken VersionedLayerClientImpl::PublishToBatch(
     const model::Publication& pub,
     const model::PublishPartitionDataRequest& request,
     PublishPartitionDataCallback callback) {
-  std::string publication_id = pub.GetId().value_or("");
-  if (publication_id.empty()) {
-    callback(client::ApiError(client::ErrorCode::InvalidArgument,
-                              "Invalid publication", true));
-    return {};
-  }
+  auto catalog = catalog_;
+  auto settings = settings_;
+  auto catalog_settings = catalog_settings_;
+
+  auto publish_task = [catalog, settings, pub, request, catalog_settings](
+                          client::CancellationContext context) mutable
```

Comment:
`CatalogSettings::GetLayerSettings()` isn't a const method.
I was thinking its because it can write to the cache, but `cache_` is shared_ptr so I can make the method const and remove that mutable.